cookiecutter/migrate.py

```diff
@@ -22,203 +22,190 @@
+            content = file.read_text(encoding="utf-8")
+            if action_pattern in content:
+                workflow_file = file
+                print(f"Found existing workflow using {action_pattern}: {file}")
+                break
+
+    if workflow_file is None:
+        workflow_file = workflow_dir / "auto-dependabot.yaml"
+
+    file_exists = workflow_file.exists()
+    if file_exists:
+        existing_content = workflow_file.read_text(encoding="utf-8")
+        # Check for auto-merge parameter in the with: section
+        if (
+            "auto-merge: 'true'" in existing_content
+            or 'auto-merge: "true"' in existing_content
+        ):
+            print(f"Workflow file {workflow_file} already has auto-merge parameter.")
+            return
+        print(f"Workflow file {workflow_file} missing auto-merge parameter.")
+        print("Updating existing workflow file...")
+
+    workflow_file.write_text(workflow_content, encoding="utf-8")
+    action = "Updated" if file_exists else "Created"
+    print(f"{action} Dependabot auto-merge workflow at {workflow_file}")
+
+
+def disable_codeowners_review_requirement() -> None:  # noqa: C901
+    """Disable CODEOWNERS review requirement in GitHub repository ruleset."""
+    import json  # pylint: disable=import-outside-toplevel
 
-    if has_filterwarnings and not has_w_flags:
-        print(
-            f"The file {path} already has a `filterwarnings` section and has no "
-            "-W flags in `addopts`, it is probably already migrated."
-        )
-    elif has_filterwarnings and has_w_flags:
-        print(
-            f"The file {path} already has a `filterwarnings` section, but also "
-            f"has -W flags in `addopts` ({' '.join(w_flags)!r}), it looks like "
-            "it is half-migrated, you should probably migrate it manually. Avoid using -W "
-            "flags in `addopts` if there is a `filterwarnings` section."
+    try:
+        result = subprocess.run(
+            ["gh", "api", "repos/:owner/:repo", "--jq", ".default_branch"],
+            capture_output=True,
+            text=True,
+            check=True,
         )
-    if not addopts_found:
-        print(f"No 'addopts' found in {path}.")
-
-    manual_step(
-        f"No changes done to {path}. "
-        "Please double check no manual steps are required."
-    )
-
-
-def migrate_mkdocs_yaml(file_path: Path) -> None:
-    """Migrate the mkdocs.yml file to fix the `paths` key location."""
-    if not file_path.is_file():
-        manual_step(f"File {file_path} does not exist, skipping automatic migration.")
+        default_branch = result.stdout.strip()
+        print(f"Default branch: {default_branch}")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to get default branch: {e}")
+        print("Skipping CODEOWNERS review requirement update.")
+        print("You may need to manually disable the CODEOWNERS review requirement.")
```

Comment:
I would remind the user that this requires admin permissions, so they don't pull their hair trying to find the option. Since you have the GitHub URL, you can even provide a URL to go look for the rulesets: `https://github.com/{org}}/{repo}/settings/rules`